src/Rings/mpoly-affine-algebras.jl

```diff
@@ -38,15 +38,20 @@ and $I$ is a homogeneous ideal of $R$ when we grade $R$ according to the corresp
 weighted degree, then $p/q$ represents the Hilbert series of $A$ as a rational function
 with denominator $q = (1-t^{w_1})\cdots (1-t^{w_n})$. 
 
-See also `hilbert_series_reduced(A::MPolyQuo)`.
+See also `hilbert_series_reduced`.
 """
-function hilbert_series(A::MPolyQuo)
+function hilbert_series(A::Union{MPolyRing, MPolyQuo})
+   if typeof(A) == FmpqMPolyRing
+      Zt, t = ZZ["t"]
+      return (one(parent(t)), (1-t)^(ngens(A)))
+   end
    H = HilbertData(A.I)
    return hilbert_series(H,1)
 end
```

Comment:
@wdecker so I don't want to annoy you, and I am fine with merging this, but let me point out that what you do here is "not the Julia way". There, we should have two methods, one for each type. This allows Julia to produce better code, and is simply more in line with usual conventions there. So, in this example, it'd look like this (assuming I understand everything correctly here):
```suggestion
function hilbert_series(A::FmpqMPolyRing)
      Zt, t = ZZ["t"]
      return (one(parent(t)), (1-t)^(ngens(A)))
end
function hilbert_series(A::Union{MPolyRing, MPolyQuo})
   H = HilbertData(A.I)
   return hilbert_series(H,1)
end
```

This assumes you really wanted to just treat `FmpqMPolyRing` in a special way, but not e.g. `FmpzMPolyRing` or `NmodMPolyRing` or .... So perhaps you also meant this?

```suggestion
function hilbert_series(A:: MPolyRing)
      Zt, t = ZZ["t"]
      return (one(parent(t)), (1-t)^(ngens(A)))
end
function hilbert_series(A::MPolyQuo)
   H = HilbertData(A.I)
   return hilbert_series(H,1)
end
```

Reply:
Of course I meant MPolyRing rather than FmpqMPolyRing, this was a copy and paste mistake. Thanks for catching it. With regard to using Union{MPolyRing, MPolyQuo}: I will come back to this later later in the more general context of graded modules. For the time being, I would appreciate if someone could merge this so that I can continue working, building on new work by Delphine and Hans. I am greatful for all the comments which help me to get a better understanding of what I do, but waiting and waiting and waiting for a merge is really annoying.